First edition of the Swap Team Racks-spell

LudvigStridsberg opened the pull request:
Added a file containing the spell and "required" it in the spell-index-file.

LudvigStridsberg committed:
First edition of the Swap Team Racks-spell

Added a file containing the spell and "required" it in the spell-index-file.

LudvigStridsberg committed:
Second edition of the Half Station Salary-spell

Changed the spell's module.exports-part so that it actually works.

trumtomte commented on spells/swap-team-rack.js:
```diff
@@ -0,0 +1,40 @@
+function swapTeamRack(payload){
+
+    // Unpack the values
+    [station, team] = payload;
```
This should probably be `const { station, team } = payload;`?

trumtomte commented on spells/swap-team-rack.js:
```diff
@@ -0,0 +1,40 @@
+    [station, team] = payload;
+
+    // Get the player's id
+    let playerId = context.id;
```
Should be `context.id()`

trumtomte commented on spells/swap-team-rack.js:
```diff
@@ -0,0 +1,40 @@
+    let player = context.getPlayerState(playerId);
+
+    // Get the player's team
+    let playerTeam = player.team;
```
Getting the player team should most likely be done by:

``` js
const playerId = context.id();
const game = context.getGameState();
const player = game.players[playerId];
const teamIndex = player.team;
```

trumtomte replied:
Since the team index is stored within the player (inside of a game) not inside the player state (which is the global state on the server **not** coupled with a game).

trumtomte reviewed (changes requested):
Added some comments.

LudvigStridsberg committed:
Second version of the Swap Team Racks-spell

Implemented Sebastian's requested changes and added some error-checking.
Now the values are unpacked correctly, it calls the function and it get's the player's team-id correctly. I also removed some unnecessary variable-assignments.

Co-Authored-By: Sebastian Bengtegård <sebastianbengtegard@gmail.com>

trumtomte commented on spells/swap-team-rack.js:
```diff
@@ -1,40 +1,53 @@
-    // Update the gamestate
-    context.updateGameState(gameState, gameState.id);
+    // Update the gamestate (which includes the racks)
+    context.updateGameState(game, game.id);
```
You can actually omit the second argument (`game.id`) since we're updating the game the player is already inside of.

trumtomte commented on spells/swap-team-rack.js:
```diff
@@ -1,40 +1,53 @@
 
     // Broadcast the event
-    context.broadcastToGame('rack:swap', {station, playerTeam, team}, gameState.id);
+    context.broadcastToGame('rack:swap', {station, playerTeamIndex, team}, game.id);
```
Change the payload to `{ station, teamOne, teamTwo }`, since it will be broadcasted to everyone within i game "player" could mean anyone.

trumtomte reviewed (changes requested):
Some small comments.

LudvigStridsberg committed:
Third version of Swap Team Racks-spell

Removed unnessecary arguments and renamed constants for easier understanding and concistency.

Co-Authored-By: Sebastian Bengtegård <sebastianbengtegard@gmail.com>

trumtomte reviewed (approved):
Great job!

Commit:
Merge branch 'main' into Spell_Swap_Team_Rack
